tutorials/02a_RunningREMINDLocally.md

```diff
@@ -1,10 +1,146 @@
+
+Download & store them in a folder of your choice. The corresponding path is provided to REMIND via environment variable. Add to your local `~/.Renviron` file (in linux, found in your home directory, in windows, found at `C:\Users\<your windows username>\Documents\.Renviron`):
+
+```bash
+# REMIND data setup
+# If the input data is spread over multiple directories,
+# separate them with a : on linux, a ; on windows
+REMIND_repos_dirs="/my/first/path:/my/second/path"
+# On windows, it would look something like
+# REMIND_repos_dirs="C:\Users\<your windows username>\REMIND;D:\REMIND"
+# depending on your used file paths
+```
+
+You do not have to unpack tar-archives as this will be done during the start-up of your REMIND run.
+
+## Start a run
+
+In a terminal session change directory to the location where REMIND was cloned via `cd <REMIND_DIR>` and start a run as you would do on the cluster. As a linux user, type
+
+```bash
+Rscript start.R
+```
+
+or as a Windows user, type
+
+```PowerShell
+Rscript.exe .\start.R
+```
+
+## Known Issues & Further Debugging
+
+### MAGCFG_STORE File Path Resolution Fails in `prepare.R`
+
+*Telltale sign* Function `prepare()` in `prepare.R` is unable to resolve the file path to `MAGCFG_STORE`, ie. the location where MAGICC configurations are stored. The error message reads:
+
+```
+Error in prepare() :
+  ERROR in MAGGICC configuration: Could not find file  ./core/magicc/MAGCFG_STORE/MAGCFG_USER_OLDDEFAULT.CFG
+```
+
+This bug might be specific to Windows systems and is still under investigation. However, there is a
+
+*Solution* Locate lines
+
+```R
+magcfgFile = paste0('./magicc/MAGCFG_STORE/','MAGCFG_USER_',toupper(cfg$gms$cm_magicc_config),'.CFG')
+[...]
+system(paste0('cp ',magcfgFile,' ','./magicc/MAGCFG_USER.CFG'))
+```
+
+in file `<REMIND_DIR>/scripts/start/prepare.R` (around line numbers 680ff). Replace them with
+
+```R
+magcfgFile = normalizePath(paste0('../../core/magicc/MAGCFG_STORE/','MAGCFG_USER_',toupper(cfg$gms$cm_magicc_config),'.CFG'))
+[...]
+system(paste0('cp ',magcfgFile,' ','../../core/magicc/MAGCFG_USER.CFG'))
+```
+
+and then restart your run.
+
+### `renv` Confuses Itself
+
+*Telltale sign* Rscript fails at the beginning of a run printing
+
+```
+Rscript : Error in eval(quote({ : object '..version..' not found
+```
+
+This occurs usually after an initial run has completed and seems to only affect people running REMIND with R-version `4.1.X`. The
+
+*Solution* is to checkout the original version of `<REMIND_DIR>/renv/activate.R` via
+
+```bash
+git checkout ./renv/activate.R
+```
+
+then restart your run.
+
+### Support RSE with Extensive Log Files
+
+In case your REMIND run fails and you wish to be supported by REMIND RSE in getting it to run properly, please help us by recording the logging output of the run. On linux, start your run via
+
+```bash
+Rscript start.R &>> ~/failed_REMIND_run.log
+```
```

Comment:
Why not to a file in the local directory?

Reply:
The log-file does not belong to the REMIND code base, nor is it relevant for REMIND execution. It is a file specific to the REMIND user. Rather than locating it amongst the REMIND files, I therefore propose to place it in their home directory

Reply:
There can be all kind of junk in the REMIND directory not required (anymore).  That is why `.log` files are ignored by git.
I think it is weird to put a log about something somebody is working on in one directory into another directory.

Reply:
I agree with Michaja, it is common practice to dump all the logs into the REMIND folder, so let's stick to that.

Reply:
Done